Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2018-08-13
Initial Package Version: 5.11.1
Upstream Status: Unknown
Origin: Fedora
Description: Fixes build with glibc-2.28.

diff -Naur a/qtbase/mkspecs/linux-g++/qplatformdefs.h b/qtbase/mkspecs/linux-g++/qplatformdefs.h
--- a/qtbase/mkspecs/linux-g++/qplatformdefs.h	2018-06-15 08:29:31.000000000 +0100
+++ b/qtbase/mkspecs/linux-g++/qplatformdefs.h	2018-08-13 04:04:57.168980072 +0100
@@ -72,7 +72,9 @@
 #include <sys/time.h>
 #include <sys/shm.h>
 #include <sys/socket.h>
+#if 0
 #include <sys/stat.h>
+#endif
 #include <sys/wait.h>
 #include <netinet/in.h>
 
diff -Naur a/qtbase/src/corelib/io/qfilesystemengine_unix.cpp b/qtbase/src/corelib/io/qfilesystemengine_unix.cpp
--- a/qtbase/src/corelib/io/qfilesystemengine_unix.cpp	2018-06-15 08:29:31.000000000 +0100
+++ b/qtbase/src/corelib/io/qfilesystemengine_unix.cpp	2018-08-13 04:04:57.168980072 +0100
@@ -50,7 +50,9 @@
 #include <pwd.h>
 #include <stdlib.h> // for realpath()
 #include <sys/types.h>
+#if 0
 #include <sys/stat.h>
+#endif
 #include <unistd.h>
 #include <stdio.h>
 #include <errno.h>
@@ -91,7 +93,9 @@
 #  include <sys/syscall.h>
 #  include <sys/sendfile.h>
 #  include <linux/fs.h>
+#if 0
 #  include <linux/stat.h>
+#endif
 
 // in case linux/fs.h is too old and doesn't define it:
 #ifndef FICLONE
@@ -105,13 +109,13 @@
 #    undef SYS_renameat2
 #    undef SYS_statx
 #    undef STATX_BASIC_STATS
-#  else
-#    if !QT_CONFIG(renameat2) && defined(SYS_renameat2)
+#  else 
+#    if 0 && !QT_CONFIG(renameat2) && defined(SYS_renameat2)
 static int renameat2(int oldfd, const char *oldpath, int newfd, const char *newpath, unsigned flags)
 { return syscall(SYS_renameat2, oldfd, oldpath, newfd, newpath, flags); }
 #    endif
 
-#    if !QT_CONFIG(statx) && defined(SYS_statx)
+#    if 0 && !QT_CONFIG(statx) && defined(SYS_statx)
 static int statx(int dirfd, const char *pathname, int flag, unsigned mask, struct statx *statxbuf)
 { return syscall(SYS_statx, dirfd, pathname, flag, mask, statxbuf); }
 #    elif !QT_CONFIG(statx) && !defined(SYS_statx)
